nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py

```diff
@@ -428,10 +441,10 @@ def inference_step(self, dataloader_iter, batch_idx, mode, dataloader_idx=0):
         if mode == 'validation':
             if type(self.trainer.val_dataloaders) == list and len(self.trainer.val_dataloaders) > 1:
                 # super().validation_step appends just loss to self.validation_step_outputs, replace the last appended loss with the outputs dict
-                self.validation_step_outputs[dataloader_idx][-1] = outputs
+                self.validation_step_outputs[dataloader_idx].append(outputs)
```

Comment:
why change this? what if we use multiple dataloaders with val (once the issue is resolved on PTL end)?

Reply:
nvm, this still indexes correctly